src/backend/application_server.py

```diff
@@ -0,0 +1,133 @@
+
+"""
+Module defining our flask application server.
+"""
+
+import os
+import logging
+from PIL import Image
+from pathlib import Path
+import torch
+import torchvision.transforms as transforms
+from flask import Flask, flash, request, redirect, render_template,\
+                  send_from_directory
+from werkzeug.utils import secure_filename
+from gan import Generator
+from torchvision.utils import save_image
+
+UPLOAD_FOLDER = Path.cwd() / 'uploads/'
+Path(UPLOAD_FOLDER).mkdir(exist_ok=True)
+RESPONSE_FOLDER = Path.cwd() / 'response/'
+Path(RESPONSE_FOLDER).mkdir(exist_ok=True)
+ALLOWED_EXTENSIONS = {'bmp', 'png', 'jpg', 'jpeg', 'ppm', 'pgm', 'tif'}
+
+app = Flask(__name__)
+app.secret_key = b'MBWUdbxX;>]vrTL'
+app.config['MAX_CONTENT_LENGTH'] = 50 * 1024 * 1024
+app.config['UPLOAD_FOLDER'] = UPLOAD_FOLDER
+app.config['RESPONSE_FOLDER'] = RESPONSE_FOLDER
+model = Generator(9)
+transform = None
+
+
+def init_inference(model_path):
+    """
+    Loads model from file and initialises image transformation.
+    :param model_path:
+    :return:
+    """
+    global model
+    global transform
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+    model.load_state_dict(torch.load(model_path, map_location=device))
+    image_size = (256, 256)
+    transform = transforms.Compose([
+        transforms.Resize(image_size),
+        transforms.CenterCrop(image_size),
+        transforms.ToTensor(),
+        transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))
+    ])
+
+
+def allowed_file(filename):
+    """
+    Given filename returns whether file is of allowed file type.
+    :param filename:
+    :return:
+    """
+    return '.' in filename and \
+           filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS
+
+
+@app.route('/', methods=['GET', 'POST'])
```

Comment:
Is the 'GET' behaviour defined implicitly/by the framework or has it been forgotten?
Can't find any code referring to it.